src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/CacheHealthDecider.java

```diff
@@ -16,146 +16,188 @@
-                return;
-            }
-
-            final HotClusterSummary clusterSummary = flowUnit.getSummary();
-
-            clusterSummary
-                    .getHotNodeSummaryList()
-                    .forEach(
-                            hotNodeSummary -> {
-                                final NodeKey esNode =
-                                        new NodeKey(hotNodeSummary.getNodeID(), hotNodeSummary.getHostAddress());
-                                for (final HotResourceSummary resource :
-                                        hotNodeSummary.getHotResourceSummaryList()) {
-                                    decision.addAction(
-                                            computeBestAction(esNode, resource.getResource().getResourceEnum()));
-                                }
-                            });
-        }
+  private static final Logger LOG = LogManager.getLogger(CacheHealthDecider.class);
+  public static final String NAME = "cacheHealthDecider";
+
+  private final FieldDataCacheClusterRca fieldDataCacheClusterRca;
+  private final ShardRequestCacheClusterRca shardRequestCacheClusterRca;
+  private final ImmutableMap<ResourceEnum, BaseClusterRca> cacheTypeBaseClusterRcaMap;
+
+  private double fieldDataCacheSizeUpperBound;
+  private double shardRequestCacheSizeUpperBound;
+
+  List<ResourceEnum> modifyCacheActionPriorityList = new ArrayList<>();
+  private int counter = 0;
+
+  public CacheHealthDecider(
+      final long evalIntervalSeconds,
+      final int decisionFrequency,
+      final FieldDataCacheClusterRca fieldDataCacheClusterRca,
+      final ShardRequestCacheClusterRca shardRequestCacheClusterRca) {
+    super(evalIntervalSeconds, decisionFrequency);
+    configureModifyCacheActionPriority();
+
+    this.fieldDataCacheClusterRca = fieldDataCacheClusterRca;
+    this.shardRequestCacheClusterRca = shardRequestCacheClusterRca;
+    this.cacheTypeBaseClusterRcaMap =
+        ImmutableMap.<ResourceEnum, BaseClusterRca>builder()
+            .put(ResourceEnum.SHARD_REQUEST_CACHE, shardRequestCacheClusterRca)
+            .put(ResourceEnum.FIELD_DATA_CACHE, fieldDataCacheClusterRca)
+            .build();
+
+    this.fieldDataCacheSizeUpperBound = CacheDeciderConfig.DEFAULT_FIELD_DATA_CACHE_UPPER_BOUND;
+    this.shardRequestCacheSizeUpperBound = CacheDeciderConfig.DEFAULT_SHARD_REQUEST_CACHE_UPPER_BOUND;
+  }
+
+  @Override
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public Decision operate() {
+    Set<InstanceDetails.Id> impactedNodes = new HashSet<>();
+
+    Decision decision = new Decision(System.currentTimeMillis(), NAME);
+    counter += 1;
+    if (counter < decisionFrequency) {
+      return decision;
     }
+    counter = 0;
 
-    private void configureActionPriority() {
-        // TODO: Input from user configured yml
-        this.actionsByUserPriority.add(ModifyCacheCapacityAction.NAME);
+    for (final ResourceEnum cacheType : modifyCacheActionPriorityList) {
```

Comment:
does that mean that cache decider will always pick shard request cache to scale up?

Reply:
Yes, presently always shard request cache is given higher priority in modify decider. 
Based on performance testing, we can change this.